extensions/grub-arm.sh

```diff
@@ -0,0 +1,101 @@
+	export BOOTSIZE=0                                                            # No separate /boot when using UEFI.
+	export CLOUD_INIT_CONFIG_LOCATION="${CLOUD_INIT_CONFIG_LOCATION:-/boot/efi}" # use /boot/efi for cloud-init as default when using Grub.
+	export EXTRA_BSP_NAME="${EXTRA_BSP_NAME}-grub"                               # Unique bsp name.
+	export UEFI_GRUB_TARGET_BIOS=""                                              # Target for BIOS GRUB install, set to i386-pc when UEFI_ENABLE_BIOS_AMD64=yes and target is amd64
+	local uefi_packages="efibootmgr efivar"				             # Use growroot, add some efi-related packages
+	uefi_packages="os-prober grub-efi-${ARCH}-bin ${uefi_packages}"              # This works for Ubuntu and Debian, by sheer luck; common for EFI and BIOS
+
+	[[ "${ARCH}" == "arm64" ]] && export uefi_packages="${uefi_packages} grub-efi-${ARCH}"
+	[[ "${ARCH}" == "arm64" ]] && export UEFI_GRUB_TARGET="arm64-efi" # Default for arm64-efi
+
+	DISTRO_KERNEL_PACKAGES=""
+	DISTRO_FIRMWARE_PACKAGES=""
+
+	export PACKAGE_LIST_BOARD="${PACKAGE_LIST_BOARD} ${DISTRO_FIRMWARE_PACKAGES} ${DISTRO_KERNEL_PACKAGES} ${uefi_packages}"
+
+}
+
+pre_umount_final_image__install_grub() {
+	configure_grub
+	local chroot_target=$MOUNT
+	display_alert "Installing bootloader" "GRUB" "info"
+
+	# getting rid of the dtb package, if installed, is hard. for now just zap it, otherwise update-grub goes bananas
+	mkdir -p "$MOUNT"/boot/efi/dtb
+	cp -r "$MOUNT"/boot/dtb/* "$MOUNT"/boot/efi/dtb/
+	cp -r ${SRC}/packages/blobs/splash/grub.png "$MOUNT"/boot/grub
+
+	# add config to disable os-prober, otherwise image will have the host's other OSes boot entries.
+	cat <<- grubCfgFragHostSide >> "${MOUNT}"/etc/default/grub.d/99-armbian-host-side.cfg
+		GRUB_DISABLE_OS_PROBER=true
+	grubCfgFragHostSide
+
+	# copy Armbian GRUB wallpaper
+	mkdir -p "${MOUNT}"/usr/share/images/grub/
+	cp "${SRC}"/packages/blobs/splash/grub.png "${MOUNT}"/usr/share/images/grub/wallpaper.png
+
+	# Mount the chroot...
+	mount_chroot "$chroot_target/" # this already handles /boot/efi which is required for it to work.
+
+	sed -i '/devicetree/c echo' "$MOUNT"/etc/grub.d/10_linux >>"${DEST}"/"${LOG_SUBPATH}"/grub-n.log 2>&1
+
+	local install_grub_cmdline="update-grub && grub-install --verbose --target=${UEFI_GRUB_TARGET} --no-nvram --removable" # nvram is global to the host, even across chroot. take care.
+	display_alert "Installing GRUB EFI..." "${UEFI_GRUB_TARGET}" ""
+	chroot "$chroot_target" /bin/bash -c "$install_grub_cmdline" >> "$DEST"/"${LOG_SUBPATH}"/install.log 2>&1 || {
+		exit_with_error "${install_grub_cmdline} failed!"
+	}
+
+	# Remove host-side config.
+	rm -f "${MOUNT}"/etc/default/grub.d/99-armbian-host-side.cfg
+
+	local root_uuid
+	root_uuid=$(blkid -s UUID -o value "${LOOP}p${uefipart}") # get the uuid of the root partition, this has been transposed
```

Comment:
This I guess is the main change from the original `grub`. I'll investigate later and we re-unify after merge.

Reply:
> This I guess is the main change from the original `grub`. I'll investigate later and we re-unify after merge.

Again a mistake, this is only a small part of the necessary changes :)

Reply:
Sure, if there are enough we'll leave your cloned extension in place. Don't worry